### PR TITLE
[dvs] Refactor ACL tests to use polling interface to access redis

### PR DESCRIPTION
**Pull request by daall**
* Introduce new DVSDatabase abstraction and SONiC redis fixtures
* Refactor ACL tests to use redis fixtures

Signed-off-by: Danny Allen <daall@microsoft.com>

**Why I did it**
* To help make the vs tests more reliable by not relying on arbitrary wait times
* To make the vs tests faster by reducing the wait time (acl tests run in ~50% of the time now)

**How I verified it**
```
daall@rainier:~/dev/sonic-swss/tests$ sudo pytest -s -v --dvsname=vs test_acl.py
=========================================================================================== test session starts ============================================================================================
platform linux2 -- Python 2.7.17, pytest-4.6.9, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /home/daall/dev/sonic-swss/tests
plugins: flaky-3.6.1
collected 24 items                                                                                                                                                                                         

test_acl.py::TestAcl::test_AclTableCreation PASSED
test_acl.py::TestAcl::test_AclRuleL4SrcPort PASSED
test_acl.py::TestAcl::test_AclRuleInOutPorts PASSED
test_acl.py::TestAcl::test_AclRuleInPortsNonExistingInterface PASSED
test_acl.py::TestAcl::test_AclRuleOutPortsNonExistingInterface PASSED
test_acl.py::TestAcl::test_AclTableDeletion PASSED
test_acl.py::TestAcl::test_V6AclTableCreation PASSED
test_acl.py::TestAcl::test_V6AclRuleIPv6Any PASSED
test_acl.py::TestAcl::test_V6AclRuleIPv6AnyDrop PASSED
test_acl.py::TestAcl::test_V6AclRuleIpProtocol PASSED
test_acl.py::TestAcl::test_V6AclRuleSrcIPv6 PASSED
test_acl.py::TestAcl::test_V6AclRuleDstIPv6 PASSED
test_acl.py::TestAcl::test_V6AclRuleL4SrcPort PASSED
test_acl.py::TestAcl::test_V6AclRuleL4DstPort PASSED
test_acl.py::TestAcl::test_V6AclRuleTCPFlags PASSED
test_acl.py::TestAcl::test_V6AclRuleL4SrcPortRange PASSED
test_acl.py::TestAcl::test_V6AclRuleL4DstPortRange PASSED
test_acl.py::TestAcl::test_V6AclTableDeletion PASSED
test_acl.py::TestAcl::test_InsertAclRuleBetweenPriorities PASSED
test_acl.py::TestAcl::test_RulesWithDiffMaskLengths PASSED
test_acl.py::TestAcl::test_AclRuleIcmp PASSED
test_acl.py::TestAcl::test_AclRuleIcmpV6 PASSED
test_acl.py::TestAcl::test_AclRuleRedirectToNextHop PASSED
test_acl.py::TestAclRuleValidation::test_AclActionValidation PASSED

======================================================================================= 24 passed in 119.13 seconds ========================================================================================
```

### CODE REVIEW

File: tests/dvslib/dvs_database.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+                key (str): The key that maps to the entry.
+                entry (Dict[str, str]): A set of key-value pairs to be stored.
+        """
+
+        table = swsscommon.Table(self.db_connection, table_name)
+        formatted_entry = swsscommon.FieldValuePairs(entry.items())
+        table.set(key, formatted_entry)
+
+    def get_entry(self, table_name, key,
+                  polling_config=DEFAULT_POLLING_CONFIG):
```
Comment (yxieca) [inline tests/dvslib/dvs_database.py]: This function is doing wait_for_entry(), maybe you should name the function accordingly?

This comment is applicable to other polling functions too.

Is there a way to disable polling and just do a one shot check? If that is the case, then maybe name the function like wait_entry_until()?

File: tests/dvslib/dvs_database.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+                key (str): The key that maps to the entry being removed.
+        """
+
+        table = swsscommon.Table(self.db_connection, table_name)
+        table._del(key)  # pylint: disable=protected-access
+
+    def is_entry_empty(self,
+                       table_name,
+                       key,
+                       polling_config=DEFAULT_POLLING_CONFIG):
```
Comment (yxieca) [inline tests/dvslib/dvs_database.py]: Want to call out this function specifically. It gives a sense of Boolean check but has a polling implemented.

Comment (daall) [reply tests/dvslib/dvs_database.py]: That's a good point, I went ahead and updated it. Let me know if the new names make sense!

File: tests/dvslib/dvs_database.py (hunk 1 above)
Comment (daall) [reply tests/dvslib/dvs_database.py]: > Is there a way to disable polling and just do a one shot check? If that is the case, then maybe name the function like wait_entry_until()?

You could do that pretty easily by zeroing all the polling params and turning off strict checking. I could make a default for that (like `DISABLE_POLLING_CONFIG`), or even provide a set of wrappers for one-shot checking, I'm just not sure how useful that will be.

File: tests/dvslib/dvs_database.py [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+        """
+
+        table = swsscommon.Table(self.db_connection, table_name)
+
+        def _accessor():
+            (status, fv_pairs) = table.get(key)
+
+            status = expect_entry == status
+
+            if status and expect_entry:
```
[the remark this answers is not in the document]
Comment (daall) [reply tests/dvslib/dvs_database.py]: Yeah that could be helpful, I'll add that!

**Comment by daall** [comment]
retest this please

**Comment by yxieca** [comment]
retest this please

**Comment by daall** [comment]
retest this please

**Pull request merged**